test/test_nestedtensor.py

```diff
@@ -1646,6 +1646,23 @@ def test_nested_tensor_mul_in_place(self, device, dtype):
             lambda: vector.mul_(nt1),
         )
 
+    @onlyCPU
+    @skipMeta
+    def test_nested_tensor_cumsum(self, device):
```

Comment:
This test is good for the simple case, but in general we try to cover the cross product of `(eager, compile) x (forward, backward) x njt dim x (contig, non-contig with holes, non-contig transposed) x operating on (batch dim, ragged dim, other dim)`, etc.

To get this relatively easily, you can use the auto-generated OpInfo tests in `test/test_nestedtensor.py`. This will get you compile and eager tests for forward & backward run over a set of SampleInputs. SampleInputs for `cumsum()` can be defined in `torch/testing/_internal/opinfo/nested.py` and there are helpers there to make generating different types of NJTs easy. Lmk if you want more detail on this and I'm happy to help.